File: mobile/sub_manager/lib/domain/entities/subscription.dart
class Subscription {
  final int id;
  final String name;
  final String category;
  final double price;
  final double yearCost;
  final bool isActive;
  final DateTime createdAt;
  final int daysBeforeNextPayment;
  final DateTime paymentDate;
  final String userId;

  const Subscription({
    required this.id,
    required this.name,
    required this.category,
    required this.price,
    required this.yearCost,
    required this.isActive,
    required this.createdAt,
    required this.daysBeforeNextPayment,
    required this.paymentDate,
    required this.userId,
  });

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is Subscription && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;

  @override
  String toString() => 'Subscription(id: $id, name: $name, price: $price)';
}

File: backend/SubManager.Domain/Entities/Subscription.cs
using SubManager.Domain.IdentityEntities;
using SubManager.Domain.Validation;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SubManager.Domain.Entities
{
    public class Subscription
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public required string Name { get; set; }

        [ForeignKey(nameof(Category))]
        public int? CategoryId { get; set; }

        [Required]
        [Range(typeof(decimal), "0", "100000")]
        public decimal Price { get; set; }

        [Required]
        [Range(1, 31)]
        public int PaymentDay { get; set; }

        public bool IsActive { get; set; }

        [Required]
        public Guid UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public virtual ApplicationUser User { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public Category? Category { get; set; }
    }
}
